### ISSUE TITLE
[MPMD-283] Create a real aggregate goal

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-pmd-plugin#616]
**[Andreas Dangel](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=adangel)** opened **[MPMD-283](https://issues.apache.org/jira/browse/MPMD-283?redirect=false)** and commented

PMD needs to have the project compiled, before it can analyze it. This is required to support typeresolution.

For Multi-modules projects, you can currently set the "aggregate" parameter, to create an aggregated report. However, this is executed at the root project _before_ all other modules have been built, leading to a potentially inaccurate PMD result and false positives.

A separate goal should be introduced, that can execute the compile lifecycle prior to executing PMD (similar what javadoc:aggregate does).

 

```java
@Mojo( name = "aggregate", aggregator = true, requiresDependencyResolution = ResolutionScope.TEST )
@Execute( phase = LifecyclePhase.COMPILE )
public class AggregatorPMDReport {
...
```

---

**Issue Links:**
- [MPMD-277](https://issues.apache.org/jira/browse/MPMD-277) Plugin tries to download local submodules from repo

**Remote Links:**
- [GitHub Pull Request #45
](https://github.com/apache/maven-pmd-plugin/pull/45)

**Comment by jira-importer** [comment]
**[Andreas Dangel](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=adangel)** commented

Fixed in https://gitbox.apache.org/repos/asf?p=maven-pmd-plugin.git;a=commit;h=5ae53f135b46102e7b749c834173cdc97678f834
